### PR TITLE
Initial flow

### PR DESCRIPTION
**Pull request by fleetwoods**
Service testing as per composer service
Fix to create the Failed events as expected

### CODE REVIEW

File: banner.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                                     mddddddm
+                                   mddddddm
+                                   mddddddm
+                                     dmmd
+
+  ___        _  _                        ___                 _
+ |   \  ___ | |(_)__ __ ___  _ _  _  _  / __| ___  _ _ __ __(_) __  ___
+ | |) |/ -_)| || |\ V // -_)| '_|| || | \__ \/ -_)| '_|\ V /| |/ _|/ -_)
+ |___/ \___||_||_| \_/ \___||_|   \_, | |___/\___||_|   \_/ |_|\__|\___|
+                                  |__/
```
Comment (cb372) [inline banner.txt]: 👍

File: circle.yml [hunk 2]
```diff
@@ -17,6 +21,7 @@ dependencies:
 test:
   post:
     - sbt docker:publishLocal
+    - sbt dockerComposeTest
```
Comment (cb372) [inline circle.yml]: Let's run this as part of the test step

File: src/test/scala/com/ovoenergy/delivery/service/ServiceTestIT.scala [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+  val emailProgressedTopic = "comms.progressed.email"
+
+  val mockServerClient = new MockServerClient("localhost", 1080)
+
+  behavior of "producer"
+
+    it should "create Failed event when authentication fails with Mailgun" taggedAs DockerComposeTag in {
+      createTopicsAndSubscribe()
+      create401MailgunResponse()
+
```
Comment (cb372) [inline src/test/scala/com/ovoenergy/delivery/service/ServiceTestIT.scala]: `Arbitrary[ComposedEmail].sample.get`

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 47](https://circleci.com/gh/ovotech/comms-delivery-service/47) at 2016-11-28 13:02 GMT.

**Pull request merged**